service/lib/agama/config.rb

```diff
@@ -20,173 +20,32 @@
-
-      products.select! do |product|
-        product["archs"].nil? ||
-          Yast2::ArchFilter.from_string(product["archs"]).match?
-      end
-
-      @products = products.each_with_object({}) do |product, result|
-        result[product["id"]] = product
-      end
+    def initialize(config_data = {})
+      @data = config_data
     end
 
-    # Whether there are more than one product
+    # Loads the configuration from a given file
     #
-    # @return [Boolean] false if there is only one product; true otherwise
-    def multi_product?
-      products.size > 1
-    end
-
-    # Returns a copy of this Object
-    #
-    # @return [Config]
-    def copy
-      logger = self.logger
-      @logger = nil # cannot dump logger as it can contain IO
-      new_config = super
-      @logger = logger
-      new_config.logger = logger
-
-      new_config
-    end
-
-    # Returns a new {Config} with the merge of the given ones
-    #
-    # @param config [Config, Hash]
-    # @return [Config] new Configuration with the merge of the given ones
-    def merge(config)
-      Config.new(simple_merge(data, config.data))
-    end
-
-    # Elements that match the current arch.
-    #
-    # @example
-    #   config.products #=>
-    #   {
-    #     "ALP-Dolomite" => {
-    #       "software" => {
-    #         "installation_repositories" => [
-    #           {
-    #             "url" => "https://updates.suse.com/SUSE/Products/ALP-Dolomite/1.0/x86_64/product/",
-    #             "archs" => "x86_64"
-    #           },
-    #           {
-    #             "url" => https://updates.suse.com/SUSE/Products/ALP-Dolomite/1.0/aarch64/product/",
-    #             "archs" => "aarch64"
-    #           },
-    #           "https://updates.suse.com/SUSE/Products/ALP-Dolomite/1.0/noarch/"
-    #         ]
-    #       }
-    #     }
-    #   }
-    #
-    #   Yast::Arch.rpm_arch #=> "x86_64"
-    #   config.arch_elements_from("ALP-Dolomite", "software", "installation_repositories",
-    #     property: :url) #=> ["https://.../SUSE/Products/ALP-Dolomite/1.0/x86_64/product/",
-    #                     #=>  "https://updates.suse.com/SUSE/Products/ALP-Dolomite/1.0/noarch/"]
-    #
-    # @param keys [Array<Symbol|String>] Config data keys of the collection.
-    # @param property [Symbol|String|nil] Property to retrieve of the elements.
-    # @param default [Object] The default value returned when the value is not
-    #   found or is not an array
-    #
-    # @return [Array]
-    def arch_elements_from(*keys, property: nil, default: [])
-      keys.map!(&:to_s)
-      elements = products.dig(*keys)
-      return default unless elements.is_a?(Array)
-
-      elements.map do |element|
-        if !element.is_a?(Hash)
-          element
-        elsif arch_match?(element["archs"])
-          property ? element[property.to_s] : element
-        end
-      end.compact
+    # @param path [String|Pathname] File path
+    def self.from_file(path)
```

Comment:
Is this really needed? If we are using it for loading data in tests, then we could define this oneliner directly in the test file.